filebeat/module/elasticsearch/audit/ingest/pipeline-json.yml

```diff
@@ -25,13 +30,13 @@ processors:
   - dot_expander:
       field: event.action
       path: elasticsearch.audit
+  - remove:
```

Comment:
how does it works now? we move a dotted `elasticsearch.audit.event.action` into `{event:{action..` 
but then we remove it?

Reply:
I've added it due to the error in the next step of the pipeline (rename `elasticsearch.audit.event.action` -> `event.action`).
Error: `message": "field [event.action] already exists"`, as a result processing of the message was stopped on this step.
<img width="2257" alt="Screenshot 2022-02-02 at 15 40 36" src="https://user-images.githubusercontent.com/28299531/152327251-abdbe0e1-2ea1-4898-b555-c22e9913a5c4.png">

> we move a dotted `elasticsearch.audit.event.action` into `{event:{action..`
but then we remove it?

hm, maybe my understanding of `dot_expander` processor is wrong. shouldn't `elasticsearch.audit.event.action` be transformed into - > `{elasticsearch: {audit : { event: {action ...`  ?
if `elasticsearch.audit.event.action` is dedotted into `{event:{action..`, why them we might need `rename` stage, defined next:
```
- rename:
      field: elasticsearch.audit.event.action
      target_field: event.action
...
```

Reply:
> hm, maybe my understanding of dot_expander processor is wrong. shouldn't elasticsearch.audit.event.action be transformed into - > {elasticsearch: {audit : { event: {action ... ?
if elasticsearch.audit.event.action is dedotted into {event:{action.., why them we might need rename stage, defined next:

- rename:
      field: elasticsearch.audit.event.action
      target_field: event.action
...

you are totally correct. I just tested this myself and it works like you described. Thanks!